app/containers/HomePage.jsx

```diff
@@ -31,6 +40,9 @@ class HomePage extends Component {
   componentDidMount() {
     this.listDevices();
   }
+  componentWillUnmount() {
+    clearInterval(this.autoRefresh);
```

Comment:
How does this work ? 
I mean ```autoRefresh = () => {
+    setInterval(() => {
+      if (result(this.props.state, 'devices.devicesList', []).length === 0) {
+        this.listDevices();
+      }
+    }, 2000);
+  }``` The function passed to setInterval is different (Its not this.autoRefresh) right ?